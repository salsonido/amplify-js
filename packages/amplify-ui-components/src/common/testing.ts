import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
