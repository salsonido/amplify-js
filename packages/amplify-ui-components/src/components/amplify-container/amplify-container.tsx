import { Component, h, Host } from '@stencil/core';

/**
 * @slot (default) - Content placed within the container
 */
@Component({
	tag: 'amplify-container',
	styleUrl: 'amplify-container.scss',
})
export class AmplifyContainer {
	render() {
		return (
			<Host>
				<slot></slot>
			</Host>
		);
	}
}
