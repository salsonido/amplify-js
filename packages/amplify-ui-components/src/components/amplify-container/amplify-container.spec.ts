import { newSpecPage } from '@stencil/core/testing';
import { AmplifyContainer } from './amplify-container';

describe('amplify-container spec:', () => {
	describe('Render logic ->', () => {
		it('should render with an empty slot', async () => {
			const page = await newSpecPage({
				components: [AmplifyContainer],
				html: `<amplify-container></amplify-container>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render with contained content when content is passed in', async () => {
			const page = await newSpecPage({
				components: [AmplifyContainer],
				html: `<amplify-container><div>My App</div></amplify-container>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
