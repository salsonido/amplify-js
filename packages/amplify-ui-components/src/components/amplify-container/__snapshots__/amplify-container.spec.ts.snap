// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-container spec: Render logic -> should render with an empty slot 1`] = `
<amplify-container>
  <!---->
</amplify-container>
`;

exports[`amplify-container spec: Render logic -> should render with contained content when content is passed in 1`] = `
<amplify-container>
  <!---->
  <div>
    My App
  </div>
</amplify-container>
`;
